File: app/src/main/java/com/omar/retromp3recorder/app/repo/SampleRateRepo.java
package com.omar.retromp3recorder.app.repo;

import com.omar.retromp3recorder.app.recorder.RecorderDefaults;
import com.omar.retromp3recorder.app.recorder.VoiceRecorder;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Observable;
import io.reactivex.subjects.BehaviorSubject;

@Singleton
public class SampleRateRepo {
    private final BehaviorSubject<VoiceRecorder.SampleRate> stateSubject = BehaviorSubject.create();

    @Inject
    SampleRateRepo(RecorderDefaults defaults){
        stateSubject.onNext(defaults.sampleRate);
    }

    public void newValue( VoiceRecorder.SampleRate bitRate) {
        stateSubject.onNext(bitRate);
    }

    public Observable<VoiceRecorder.SampleRate> observe() {
        return stateSubject;
    }
}

File: app/src/main/java/com/omar/retromp3recorder/app/state/SampleRateRepo.kt
package com.omar.retromp3recorder.app.state

import com.omar.retromp3recorder.app.modules.recording.Mp3VoiceRecorder.SampleRate
import com.omar.retromp3recorder.app.modules.recording.RecorderDefaults
import io.reactivex.Observable
import io.reactivex.subjects.BehaviorSubject
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class SampleRateRepo @Inject constructor(defaults: RecorderDefaults) {
    private val stateSubject = BehaviorSubject.create<SampleRate>()

    init {
        stateSubject.onNext(defaults.sampleRate)
    }

    fun newValue(bitRate: SampleRate) {
        stateSubject.onNext(bitRate)
    }

    fun observe(): Observable<SampleRate> {
        return stateSubject
    }
}
